### PR TITLE
[#23] more precise types

### PR DESCRIPTION
**Pull request by stereobooster**
**What**: Even more precise type signatures for Flow

**Why**: To increase Flow coverage

**How**: declared return type using parametrised types

**Checklist**:

* N/A Documentation
* [x] Tests
* [ ] Ready to be merged
      
* N/A Added myself to contributors table
      

This is not ready to merge. Just curious what do you think about this approach.

### CODE REVIEW

File: flow-typed/npm/downshift_v2.x.x.js.flow [hunk 1]
```diff
@@ -114,27 +114,97 @@ declare module downshift {
   declare export type StateChangeFunction<Item> = (
     state: DownshiftState<Item>,
   ) => StateChangeOptions<Item>
+
+  declare export type GetRootPropsReturn = {
+    role: 'combobox';
+    // 'aria-expanded': boolean;
```
Comment (stereobooster) [inline flow-typed/npm/downshift_v2.x.x.js.flow]: Flow doesn't like properties declared as strings

File: flow-typed/npm/downshift_v2.x.x.js.flow [hunk 2]
```diff
@@ -114,27 +114,97 @@ declare module downshift {
...
   }
-  declare export interface GetInputPropsOptions
-    extends React.HTMLProps<HTMLInputElement> {}
-  declare export interface GetLabelPropsOptions
-    extends React.HTMLProps<HTMLLabelElement> {}
+
+  declare type GetToggleButtonCallbacks = {
+    onMouseMove: (e: React.MouseEvent<Element>) => void;
+    onMouseDown: (e: React.MouseEvent<Element>) => void;
+    onBlur: (e: React.MouseEvent<Element>) => void;
```
Comment (stereobooster) [inline flow-typed/npm/downshift_v2.x.x.js.flow]: This is probably not `MouseEvent`

File: flow-typed/npm/downshift_v2.x.x.js.flow (hunk 1 above)
Comment (kentcdodds) [reply flow-typed/npm/downshift_v2.x.x.js.flow]: That's pretty unfortunate and surprising as well 🤔

File: flow-typed/npm/downshift_v2.x.x.js.flow (hunk 2 above)
Comment (kentcdodds) [reply flow-typed/npm/downshift_v2.x.x.js.flow]: Hmmm... I'm confused. If it's not `MouseEvent` then why is it typed to be a `MouseEvent` here?

Comment (stereobooster) [reply flow-typed/npm/downshift_v2.x.x.js.flow]: I meant: this is an error I'm aware of

File: flow-typed/npm/downshift_v2.x.x.js.flow (hunk 1 above)
Comment (RichieAHB) [reply flow-typed/npm/downshift_v2.x.x.js.flow]: I'm not seeing that in the REPL? https://flow.org/try/#0C4TwDgpgBAKhDOwoF4oG8BQUoEgBGA-AFxR4D2ZANhAIYB2A3BjgOQC2IAtHQK5t4QATixK9+QpgF8MGAMZk6iKDRJwlqNFHZcxA4SQDMUSUyA

File: flow-typed/npm/downshift_v2.x.x.js.flow (hunk 2 above)
Comment (RichieAHB) [reply flow-typed/npm/downshift_v2.x.x.js.flow]: Can this not just be typed with a `SyntheticEvent`?

**Comment by stereobooster** [comment]
I updated PR. Fixed issues mentioned above, but it feels that Flow is really slow. Not sure if this is connected to the changes in this PR or not

**Pull request merged**